src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/ClientMethodsAnalyzer.cs

```diff
@@ -279,9 +279,47 @@ private static bool IsClientMethodReturnType(ISymbolAnalysisContext context, IMe
             return false;
         }
 
+        private static bool DoReturnTypesMatch(ITypeSymbol asyncReturnType, ITypeSymbol syncReturnType)
+        {
+            if (asyncReturnType == null || syncReturnType == null)
+            {
+                return false;
+            }
+
+            if (asyncReturnType is INamedTypeSymbol asyncNamedType && asyncNamedType.IsGenericType && asyncNamedType.Name == TaskTypeName)
+            {
+                var asyncInnerType = asyncNamedType.TypeArguments.Single();
+                return SymbolEqualityComparer.Default.Equals(asyncInnerType, syncReturnType);
+            }
+
+            return false;
+        }
+
+        private bool IsAmqpBasedLibrary(INamedTypeSymbol type)
+        {
+            var namespaceName = type.ContainingNamespace.ToDisplayString();
+
+            // List of AMQP-based libraries that are exempted from this rule.
+            if (namespaceName.StartsWith("Azure.Messaging.EventHubs") ||
+                namespaceName.StartsWith("Azure.Messaging.ServiceBus") ||
+                namespaceName.StartsWith("Azure.Messaging.WebPubSub"))
+            {
```

Comment:
This is the list of AMQP libraries that are suppressing the rule, as it can be seen [here](https://grep.app/search?q=azc0004&filter[repo][0]=Azure/azure-sdk-for-net).

I am unaware if there is a more general way to check if a library is AMQP based, so I would appreciate any tips or suggestions on this.

Reply:
Actually, the analyzer should not have these exceptions on itself. For AMQP services, the suppression should be in their respectives SDKs.